### PR TITLE
feat(mdns): check if the txt items is changed when browsing

### PR DESCRIPTION
**Pull request by gytxxsy**
When browsing, it is necessary to check for new TXT records. If the TXT records are changed, an update operation should be performed.

### CODE REVIEW

File: components/mdns/mdns.c [hunk 1]
```diff
@@ -7521,6 +7521,21 @@ static mdns_browse_t *_mdns_browse_find_from(mdns_browse_t *b, mdns_name_t *name
     return b;
 }
 
+static bool is_txt_item_exist(mdns_txt_item_t txt, uint8_t txt_value_len, mdns_txt_item_t *txt_list, uint8_t *txt_value_len_list, size_t txt_count)
```
Comment (david-cermak) [inline components/mdns/mdns.c:7524]: nitpick, suggest we give it a better name...
```suggestion
static bool is_txt_item_in_list(mdns_txt_item_t txt, uint8_t txt_value_len, mdns_txt_item_t *txt_list, uint8_t *txt_value_len_list, size_t txt_count)
```

File: components/mdns/mdns.c [hunk 2]
```diff
@@ -7541,9 +7556,20 @@ static void _mdns_browse_result_add_txt(mdns_browse_t *browse, const char *insta
...
                 !_str_null_or_empty(r->service_type) && !strcasecmp(service, r->service_type) &&
                 !_str_null_or_empty(r->proto) && !strcasecmp(proto, r->proto)) {
+            bool should_update = false;
             if (r->txt) {
+                // Check if txt changed
+                if (txt_count != r->txt_count) {
+                    should_update = true;
+                } else {
+                    for (size_t txt_index = 0; txt_index < txt_count; txt_index++) {
+                        if (!is_txt_item_exist(txt[txt_index], txt_value_len[txt_index], r->txt, r->txt_value_len, r->txt_count)) {
```
Comment (david-cermak) [inline components/mdns/mdns.c:7566]: are the txt items in the list supposed to be ordered the same way as in the record? If so, maybe we can give it an index to start the next `for()` loop in the function from there. That way we'd save some iterations for most common usecases, where nothing changes and we'd have to go over the list twice.

**Review by david-cermak — APPROVED** [review]
LGTM, just two notes/questions

File: components/mdns/mdns.c (hunk 2 above)
Comment (gytxxsy) [reply components/mdns/mdns.c]: TXT entries do not require a strict order, and entries in the old list and the new list may have the same content but in a different order. Therefore, it is necessary to traverse the entire list.

File: components/mdns/mdns.c (hunk 1 above)
Comment (gytxxsy) [reply components/mdns/mdns.c]: It's a great suggestion. Thanks.

File: components/mdns/mdns.c (hunk 2 above)
Comment (david-cermak) [reply components/mdns/mdns.c]: What I meant was that perhaps a tiny optimization for the most common usecase where we receive the PTR of the same service (with the same TXT) record could be done, as in this case we'd get the TXT records in the same order.

I know that they could be arranged differently, but we can handle this common case faster if we start searching from the last outer index in the inner loop (also considering that the same service gets announced multiple times) ... but that would only be beneficial if we have like hunder(s) TXT record...

anyway, just a thought.

**Pull request merged**